### PR TITLE
VEIOSS-343 - Implement API to BTA for Payments Info

### PR DESCRIPTION
**Pull request by modivilioglu**

### CODE REVIEW

File: app/uk/gov/hmrc/iossreturns/config/DesConfig.scala [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.iossreturns.config
+
+import play.api.Configuration
+
+import javax.inject.Inject
+
+class DesConfig @Inject()(config: Configuration) {
```
Comment (DevAndyroo) [inline app/uk/gov/hmrc/iossreturns/config/DesConfig.scala]: Can we call this GetFinancialDataConfig, configs aren't shared between endpoints anymore

File: test/project/build.properties [hunk 2]
```diff
@@ -0,0 +1 @@
+sbt.version=1.6.2
```
Comment (DevAndyroo) [inline test/project/build.properties]: why do we need to define this specifically for test?

File: app/uk/gov/hmrc/iossreturns/connectors/FinancialDataConnector.scala [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+class FinancialDataConnector @Inject()(
+                                        http: HttpClient,
+                                        desConfig: DesConfig
+                                      )(implicit ec: ExecutionContext) extends Logging {
+
+  private implicit val emptyHc: HeaderCarrier = HeaderCarrier()
+  private val headers: Seq[(String, String)] = desConfig.desHeaders
+
+  private def financialDataUrl(vrn: Vrn) =
+    s"${desConfig.baseUrl}enterprise/financial-data/${vrn.name.toUpperCase()}/${vrn.value}/${desConfig.regimeType}"
```
Comment (DevAndyroo) [inline app/uk/gov/hmrc/iossreturns/connectors/FinancialDataConnector.scala]: I believe we shouldn't be using VRN for this and instead IOSSNumber and passing in an IOSS number based on the spec changes in the spec document

File: app/uk/gov/hmrc/iossreturns/connectors/FinancialDataConnector.scala [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    val url = financialDataUrl(vrn)
+    http.GET[FinancialDataResponse](
+      url,
+      queryParameters.toSeqQueryParams,
+      headers = headers
+    ).recover {
+      case e: HttpException =>
+        logger.error(s"Unexpected error response getting financial data from $url, received status ${e.responseCode}, body of response was: ${e.message}")
+        Left(
+          UnexpectedResponseStatus(e.responseCode, s"Unexpected response from DES, received status ${e.responseCode}")
```
Comment (DevAndyroo) [inline app/uk/gov/hmrc/iossreturns/connectors/FinancialDataConnector.scala]: rename this to use financial data, if we can avoid using "DES" anywhere would be good.. at somepoint this endpoint will move to IF

**Review by DevAndyroo — CHANGES_REQUESTED** [review]

File: app/uk/gov/hmrc/iossreturns/connectors/FinancialDataConnector.scala (hunk 4 above)
Comment (modivilioglu) [reply app/uk/gov/hmrc/iossreturns/connectors/FinancialDataConnector.scala:51]: Done.

File: test/project/build.properties (hunk 2 above)
Comment (modivilioglu) [reply test/project/build.properties]: Removed.

File: app/uk/gov/hmrc/iossreturns/config/DesConfig.scala (hunk 1 above)
Comment (modivilioglu) [reply app/uk/gov/hmrc/iossreturns/config/DesConfig.scala]: Sure, done.

File: app/uk/gov/hmrc/iossreturns/connectors/FinancialDataConnector.scala (hunk 4 above)
Comment (modivilioglu) [reply app/uk/gov/hmrc/iossreturns/connectors/FinancialDataConnector.scala]: Done, changed Des to FinancialData in all locations

File: app/uk/gov/hmrc/iossreturns/connectors/FinancialDataConnector.scala (hunk 3 above)
Comment (modivilioglu) [reply app/uk/gov/hmrc/iossreturns/connectors/FinancialDataConnector.scala]: Done.

File: test/project/build.properties (hunk 2 above)
Comment (modivilioglu) [reply test/project/build.properties:1]: Removed.

File: app/uk/gov/hmrc/iossreturns/config/DesConfig.scala (hunk 1 above)
Comment (modivilioglu) [reply app/uk/gov/hmrc/iossreturns/config/DesConfig.scala]: Renamed to FinancialDataConfig for it to be consistent with other namings.

**Review by DevAndyroo — DISMISSED** [review]

**Review by DevAndyroo — DISMISSED** [review]

**Review by DevAndyroo — APPROVED** [review]

**Pull request merged**